idaes_examples/notebooks/docs/scaling/scaler_workshop.ipynb

```diff
@@ -0,0 +1,1596 @@
+    "            submodel=model.control_volume.properties_out,\n",
+    "            method=\"constraint_scaling_routine\",\n",
+    "            submodel_scalers=submodel_scalers,\n",
+    "            overwrite=overwrite,\n",
+    "        )\n",
+    "        self.call_submodel_scaler_method(\n",
+    "            submodel=model.control_volume.reactions,\n",
+    "            method=\"constraint_scaling_routine\",\n",
+    "            submodel_scalers=submodel_scalers,\n",
+    "            overwrite=overwrite,\n",
+    "        )\n",
+    "\n",
+    "        # =======================================================================================\n",
+    "        # New Code\n",
+    "        # Scale control volume constraints\n",
+    "        for c in model.control_volume.component_data_objects(\n",
+    "            Constraint, descend_into=False\n",
+    "        ):\n",
+    "            self.scale_constraint_by_nominal_value(\n",
+    "                c,\n",
+    "                scheme=ConstraintScalingScheme.inverseMaximum,\n",
+    "                overwrite=overwrite,\n",
+    "            )\n",
+    "\n",
+    "        # Scale unit level constraints\n",
+    "        if hasattr(model, \"rate_reaction_constraint\"):\n",
+    "            for c in model.rate_reaction_constraint.values():\n",
+    "                self.scale_constraint_by_nominal_value(\n",
+    "                    c,\n",
+    "                    scheme=ConstraintScalingScheme.inverseMaximum,\n",
+    "                    overwrite=overwrite,\n",
+    "                )\n",
+    "        # ======================================================================================="
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "242dd26c",
+   "metadata": {},
+   "source": [
+    "Once again, let us use the ``check_scaling`` function to see how our ``Scaler`` performs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "75d11627",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_scaling()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "77fe3766",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build a new instance of the model for testing\n",
+    "test_model = build_model()\n",
+    "scaler = EquilibriumReactorScaler()\n",
+    "scaler.scale_model(test_model.fs.equil)\n",
+    "\n",
+    "# Check that the number of scaling facotrs assigned matches expectations\n",
+    "# We will assume they were set correctly\n",
+    "assert len(test_model.fs.equil.scaling_factor) == 1\n",
+    "assert len(test_model.fs.equil.control_volume.scaling_factor) == 14\n",
+    "assert len(test_model.fs.equil.control_volume.properties_in[0].scaling_factor) == 8\n",
+    "assert len(test_model.fs.equil.control_volume.properties_out[0].scaling_factor) == 9\n",
+    "assert len(test_model.fs.equil.control_volume.reactions[0].scaling_factor) == 4"
+   ]
+  },
```

Comment:
This cell isn't labeled properly, so it's showing up in the user version of the notebook.